index.js

```diff
@@ -1,480 +1,183 @@
+      await _retryAuth(_retryCount)
+    } else {
+      accessToken = response.data['access_token']
+    }
   }
+
 }
 
-/**
-* Requests an item range from the API
-*
-* Return format:
-* { data: { total: 1, entries: [ [Object] ] },
-*  url: 'https://nypl-sierra-test.iii.com/iii/sierra-api/v3/items/?limit=1&id=17292415&fields=default,fixedFields,varFields,normTitle,normAuthor,orders,locations' }
-*
-* @param  {string} itemIdStart - the bnumber of the bib you want to request
-* @param  {string} itemIdEnd - the bnumber of the bib you want to request
-* @param  {function} cb - callback
-*/
-exports.requestRangeItem = (itemIdStart, itemIdEnd, cb) => {
-  if (!exports.authorizedToken) {
-    console.error('No authorizedToken set')
-    if (cb) cb('No authorizedToken set', false)
+async function _retryAuth(_retryCount) {
+  if (_retryCount <= 3) {
+    logger.warning(`Authentication retry #${_retryCount} due to empty response from Sierra API`)
+    await delay(1000 * Math.pow(2, _retryCount - 1))
+    await authenticate(_retryCount + 1)
   } else {
-    var limit = ''
-    if (itemIdEnd === '') limit = '&limit=50'
-    var url = `${exports.credsBase}items/?id=[${itemIdStart},${itemIdEnd}]${limit}&fields=default,fixedFields,varFields`
-    console.log(url)
-    // use the bearer auth token
-    request.get(url, {
-      'timeout': 120 * 1000,
-      'auth': {
-        'bearer': exports.authorizedToken
-      }
-    },
-      (error, response, body) => {
-        if (error) console.error(error)
-        if (!response) {
-          if (cb) cb(error, false)
-          return false
-        }
-        if (response.statusCode && response.statusCode === 200) {
-          if (cb) cb(null, {data: JSON.parse(body), url: url})
-        } else {
-          if (cb) cb(body, false)
-        }
-      })
+    const error = new RetryError('Authentication')
+    logger.error(error.message)
+    throw error
   }
 }
 
-/**
-* Requests all the items of a specified bib id
-* Return format:
-* { data: { total: 2, entries: [ [Object], [Object] ] },
-*   url: [ 'https://catalog.library.org/iii/sierra-api/v3/items/?bibIds=17292415&fields=default,fixedFields,varFields&offset=0' ] }
-*
-* @param  {string} bibId - the bnumber of the bib you want to request
-* @param  {function} cb - callback
-*/
-exports.requestBibItems = (bibId, cb) => {
-  if (!exports.authorizedToken) {
-    console.error('No authorizedToken set')
-    if (cb) cb('No authorizedToken set', false)
-  } else {
-    var urls = []
-    var urlsUsed = []
-    var results = []
-    var foundAllItems = false
-
-    // make an array of the urls we are going to request
-    for (var x = 0; x < 101; x++) {
-      urls.push(`${exports.credsBase}items/?bibIds=${bibId}&fields=default,fixedFields,varFields&offset=${x * 50}`)
+async function get(path, _retryCount = 1) {
+  try {
+    await authenticate()
+    const response = await axios.get(credsBase + path, {
+      'timeout': 120 * 1000, 'auth': { 'bearer': accessToken }
```

Comment:
We sure this works? Wasn't working for me as written - I think because this usage uses `request` conventions/defaults, which are different in `axios`. See "installation-tests" comment